connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerConnectorTest.java

```diff
@@ -142,11 +174,13 @@ public void testFailureIsFinalState() {
         ctx.close();
         expectLastCall();
 
-        offsetStorageReader.close();
-        expectLastCall();
+        if (connectorType == ConnectorType.SOURCE) {
+            offsetStorageReader.close();
+            expectLastCall();
 
-        offsetStore.stop();
-        expectLastCall();
+            offsetStore.stop();
+            expectLastCall();
+        }
```

Comment:
@clolov Hi! I'm aware of the effort to transition to Mockito; I don't think this change makes that transition significantly harder. There are already examples of parameterized tests in the Connect code base that use Mockito (such as the [WorkerTest suite](https://github.com/apache/kafka/blob/cfdd567955588e134770a9145ba57800ca88313c/connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerTest.java#L200-L207)) that can be followed when we make that transition for this test suite, and all of the other logic in the changes here is trivial to translate between the two mocking libraries. I'd rather keep the changes here focused and in-scope for the ticket that they aim to address, so I would prefer not to add switching over to a new mocking library to this PR.

Reply:
I don't see this as significantly different than a lot of the other repeated sections in the test, such as the [expectations for clean startup of a non-paused connector](https://github.com/apache/kafka/blob/cfdd567955588e134770a9145ba57800ca88313c/connect/runtime/src/test/java/org/apache/kafka/connect/runtime/WorkerConnectorTest.java#L173-L183). While it might be nice to fold some of these sections into reusable methods, the changes here don't make this repetition any worse than it already is, and I'm worried that decomposing them might complicate the Mockito transition that @clolov has mentioned.

Reply:
hmm I did see the other repeated sections in the tests but since they are out of scope for the context of this PR, didn't bring them up. I guess we can live with these newer repetitions.